core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -340,27 +354,60 @@ private String generatePublicKeyCert() {
   }
 
   // Schedules task to be executed once the provided futures are complete.
-  private <T> ListenableFuture<T> whenAllComplete(
-      Callable<T> task, ListenableFuture<?>... futures) {
+  private <T> ListenableFuture<T> whenAllSucceed(Callable<T> task, ListenableFuture<?>... futures) {
     SettableFuture<T> taskFuture = SettableFuture.create();
 
     // Create a countDown for all Futures to complete.
     AtomicInteger countDown = new AtomicInteger(futures.length);
 
     // Trigger the task when all futures are complete.
-    Runnable runWhenInputsAreComplete =
-        () -> {
-          if (countDown.decrementAndGet() == 0) {
-            taskFuture.setFuture(executor.submit(task));
+    FutureCallback<Object> runWhenInputAreComplete =
+        new FutureCallback<Object>() {
+          @Override
+          public void onSuccess(@NullableDecl Object o) {
+            if (countDown.decrementAndGet() == 0) {
+              taskFuture.setFuture(executor.submit(task));
+            }
+          }
+
+          @Override
+          public void onFailure(Throwable throwable) {
+            if (!taskFuture.setException(throwable)) {
+              String msg =
+                  "Got more than one input Future Failure. Logging failures after the first";
```

Comment:
Nit: I'd leave out "Future" here, and lowercase "failure".

Reply:
IIRC, I copied verbatim from the guava line